sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1781,14 +1713,17 @@ def task_worker(self):
                     continue
 
                 if not sfp_status_helper.detect_port_in_error_status(logical_port_name, self.xcvr_table_helper.get_status_tbl(asic_index)):
-                    post_port_dom_info_to_db(logical_port_name, self.port_mapping, self.xcvr_table_helper.get_dom_tbl(asic_index), self.task_stopping_event, dom_info_cache=dom_info_cache)
-                    post_port_dom_threshold_info_to_db(logical_port_name, self.port_mapping, self.xcvr_table_helper.get_dom_threshold_tbl(asic_index), self.task_stopping_event, dom_th_info_cache=dom_th_info_cache)
-                    update_port_transceiver_status_table_hw(logical_port_name,
-                                                            self.port_mapping,
-                                                            self.xcvr_table_helper.get_status_tbl(asic_index),
-                                                            self.task_stopping_event,
-                                                            transceiver_status_cache=transceiver_status_cache)
-                    post_port_pm_info_to_db(logical_port_name, self.port_mapping, self.xcvr_table_helper.get_pm_tbl(asic_index), self.task_stopping_event, pm_info_cache=pm_info_cache)
+                    try:
+                        post_port_dom_info_to_db(logical_port_name, self.port_mapping, self.xcvr_table_helper.get_dom_tbl(asic_index), self.task_stopping_event, dom_info_cache=dom_info_cache)
+                        update_port_transceiver_status_table_hw(logical_port_name,
+                                                                self.port_mapping,
+                                                                self.xcvr_table_helper.get_status_tbl(asic_index),
+                                                                self.task_stopping_event,
+                                                                transceiver_status_cache=transceiver_status_cache)
+                        post_port_pm_info_to_db(logical_port_name, self.port_mapping, self.xcvr_table_helper.get_pm_tbl(asic_index), self.task_stopping_event, pm_info_cache=pm_info_cache)
+                    except (KeyError, TypeError):
+                        #continue to process next port since execption could be raised due to port reset, transceiver removal
+                        continue
```

Comment:
should we log something here?

Reply:
I have now added a warning message for each of the 3 functions.